src/Parallel/GlobalCache.hpp

```diff
@@ -85,10 +64,18 @@ struct get_component_if_mocked_helper {
                             ComponentToFind>;
 };
 
+template <typename... Tags>
+auto make_mutable_cache_tag_storage(
+    tuples::TaggedTuple<Tags...>&& input) noexcept {
+  return tuples::TaggedTuple<MutableCacheTag<Tags>...>(
+      std::make_tuple<typename Tags::type, std::vector<CkCallback>>(
```

Comment:
So do you mean simply adding
```
using mutable_cache_callback = CkCallback;
```
somewhere near the top of GlobalCache.hpp, and then substituting `CkCallback` -> `Parallel::mutable_cache_callback` everywhere?  Or do you mean also providing new functions that will create `mutable_cache_callback`s by wrapping some (all?) of the `CkCallback` constructors?

Note that there are some (simple) Charm things that we don't wrap, like `CkLocalBranch()`.

Reply:
Yeah, I was advocating for just the type alias as you said, rather than a detailed new class definition that wraps the functionality. It is a good point, though, that we haven't been as persistent with that design structure as I had recalled in my last comment.

In this case, I think I have the weak preference in favor of having the type-alias, because the way we use the callback doesn't have much to do with charm++ callbacks (if I understand properly, we don't give this callback to charm at all, we just use it in our code directly), it's just a reasonably close-to-hand function object that works for our purposes.

Reply:
We actually do give this callback to charm++, which is why we use charm++'s type.   The `callback.send(nullptr)` in MutableGlobalCache::mutate is a charm++ call. In contrast, the callbacks in the horizon finder and the interpolator are not charm++ things and don't need to be, so we use our own types.

Presumably here the mutate function could store something like simple_actions instead of charm++ callbacks. But we want to have a std::vector of them that is of unknown length and is filled and emptied at runtime as various things execute asynchronously, which is harder to do for simple_actions.  And charm++ callbacks already have the functionality we want.

Reply:
ah, I've misunderstood. In that case, I think it should stay as-is with the charm types. Thanks for clarifying!

Reply:
I think there is a good argument for having a "strong typedef" (a class that inherits from `CkCallback`). However, not knowing what the exact type of `CkCallback` is in different situations it could just be a some function pointer and we can't inherit from it. This idea is definitely beyond the scope of this PR, but wrapping some more of the Charm++ things in concrete types (either by actually wrapping them or by using strong typedefs) would make the code base a) more type safe and b) more independent of the details of Charm++. Both reasonable goals overall, but as I said, beyond the scope of this PR.

For @moxcodes: something similar to what you did with the NodeLock :)